riva/clients/nmt/riva_nmt_t2t_client.cc

```diff
@@ -90,6 +90,24 @@ translateBatch(
   }
 }
 
+int countWords(const std::string& text) {
+    int wordCount = 0;
+    bool insideWord = false;
+    bool lastWasSpace = true;  
+    for (char c : text) {
+        if (std::isspace(c)) {
+            insideWord = false;
+            lastWasSpace = true;
+        } else {
+            if (!insideWord && lastWasSpace) {
+                wordCount++;
+                insideWord = true;
+            }
+            lastWasSpace = false;
+        }
+    }
```

Comment:
can we simplify this logic by using a single flag? when we encounter a space after a non-space character, we increment the counter. Rest of the cases we can just skip over and do nothing.

Reply:
if we have more than 2 spaces between words, it will get miscounted as a new word. I think, if you move line 101 inside the condition at line 99, it would handle this case also.

Reply:
I think the function works fine even if there are more than two spaces between the words. The "wasSapce" flag remains true even if it encounters successive spaces so the wordCount will not increase.